modules/renter/files.go

```diff
@@ -107,6 +108,34 @@ func (f *file) uploadProgress() float64 {
 	return 100 * (float64(uploaded) / float64(desired))
 }
 
+// redundancy returns the redundancy of the least redundant chunk. A file
+// becomes available when this redundancy is >= 1. Assumes that every piece is
+// unique within a file contract. -1 is returned if the file has size 0.
+func (f *file) redundancy() float64 {
+	if f.size == 0 {
+		return -1
+	}
+	piecesPerChunk := make([]int, f.numChunks())
+	// We cannot get the redundancy if piecesPerChunk has 0 length because later
+	// in the function we use the first element of the slice.
+	if len(piecesPerChunk) == 0 {
+		build.Critical("cannot get redundancy of a file with 0 chunks")
```

Comment:
I think that this has to be changed, because it is Sia's plan to support 0 size files, which means they don't need to be uploaded, which means their redundancy is... undefined. Returning -1 or a `-` is probably the right thing to do here.